frontend/src/components/Pages/InspectionPage/InspectionTable.tsx

```diff
@@ -22,161 +22,360 @@ const StyledIcon = styled(Icon)`
+                key={inspection.missionDefinition.id}
+                inspection={inspection}
+                scheduledMissions={scheduledMissions}
+                ongoingMissions={ongoingMissions}
+                openDialog={openDialog}
+                setMissions={setSelectedMissions}
+                openScheduledDialog={openScheduleDialog}
+                navigate={navigate}
+            />
+        ))
+
+    return (
+        <>
+            <StyledTable>
+                <Table>
+                    <Table.Caption>
+                        <Typography variant="h3" style={{ marginBottom: '14px' }}>
+                            {TranslateText('Inspection Missions') + ' ' + TranslateText('for') + ' ' + deck.deckName}
+                        </Typography>
+                    </Table.Caption>
+                    <Table.Head sticky>
+                        <Table.Row>
+                            {columns.map((col) => (
+                                <Table.Cell key={col}>{TranslateText(col)}</Table.Cell>
+                            ))}
+                        </Table.Row>
+                    </Table.Head>
+                    <Table.Body>{cellValues}</Table.Body>
+                </Table>
+            </StyledTable>
+            {isScheduledDialogOpen && (
+                <AlreadyScheduledMissionDialog openDialog={openDialog} closeDialog={closeScheduleDialog} />
+            )}
+        </>
+    )
+}
+
+export function AllInspectionsTable({ inspections, scheduledMissions, ongoingMissions }: ITableProps) {
+    const { TranslateText } = useLanguageContext()
+    const [selectedMissions, setSelectedMissions] = useState<CondensedMissionDefinition[]>()
+    const [isDialogOpen, setIsDialogOpen] = useState<boolean>(false)
+    const [isScheduledDialogOpen, setIsScheduledDialogOpen] = useState<boolean>(false)
+    const [unscheduledMissions, setUnscheduledMissions] = useState<CondensedMissionDefinition[]>([])
+    const [isAlreadyScheduled, setIsAlreadyScheduled] = useState<boolean>(false)
+
+    const openDialog = () => {
+        setIsDialogOpen(true)
+    }
+
+    const openScheduleDialog = () => {
+        setIsScheduledDialogOpen(true)
+    }
+
+    const closeDialog = () => {
+        setIsAlreadyScheduled(false)
+        setSelectedMissions([])
+        setUnscheduledMissions([])
+        setIsDialogOpen(false)
+    }
+
+    const closeScheduleDialog = () => {
+        setIsScheduledDialogOpen(false)
+    }
+
+    useEffect(() => {
+        let unscheduledMissions: CondensedMissionDefinition[] = []
+        if (selectedMissions) {
+            selectedMissions.forEach((mission) => {
+                if (Object.keys(scheduledMissions).includes(mission.id) && scheduledMissions[mission.id])
+                    setIsAlreadyScheduled(true)
+                else unscheduledMissions = unscheduledMissions.concat([mission])
+            })
+            setUnscheduledMissions(unscheduledMissions)
+        }
+    }, [isDialogOpen])
```

Comment:
Introduces warning
![image](https://github.com/equinor/flotilla/assets/33518988/f5469208-18c1-44ea-9acd-9c23d79e80c4)

Reply:
We have this warning on several places and I'm honestly not sure what the fix here is

Reply:
We can fix it in another PR imo